src/main/java/cs505/grad1/sensoragg/SensorAggHashMap.java

```diff
@@ -17,12 +15,55 @@
  */
 public class SensorAggHashMap extends HashMap<Integer, SensorData> {
 
-    //WELL-BEHAVED METHODS
+    private DataIterator iterator;
```

Comment:
Looks like your type here can be the java.util.Iterator interface from the Collections framework and just bind it to the class you want with Generics